test/integration/segmentation_camera_plugin.cc

```diff
@@ -0,0 +1,351 @@
+  box2->SetLocalRotation(0, 0, 0);
+  box2->SetUserData("label", middleBoxLabel);
+  box2->SetLocalScale(unitBoxSize, unitBoxSize, unitBoxSize);
+  root->AddChild(box2);
+
+  // create box visual of the hidden box
+  ignition::rendering::VisualPtr hiddenBox = scene->CreateVisual();
+  hiddenBox->AddGeometry(scene->CreateBox());
+  hiddenBox->SetOrigin(0.0, 0.0, 0.0);
+  hiddenBox->SetLocalPosition(hiddenPosition);
+  hiddenBox->SetLocalRotation(0, 0, 0);
+  hiddenBox->SetUserData("label", hiddenLabel);
+  hiddenBox->SetLocalScale(unitBoxSize, unitBoxSize, unitBoxSize);
+  root->AddChild(hiddenBox);
+}
+
+/////////////////////////////////////////////////
+class SegmentationCameraSensorTest: public testing::Test,
+  public testing::WithParamInterface<const char *>
+{
+  // Create a Segmentation Camera sensor from a SDF and gets a image message
+  public: void ImagesWithBuiltinSDF(const std::string &_renderEngine);
+};
+
+/////////////////////////////////////////////////
+void SegmentationCameraSensorTest::ImagesWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "segmentation_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  int width = imagePtr->Get<int>("width");
+  int height = imagePtr->Get<int>("height");
+
+  // If ogre2 is not the engine, don't run the test
+  if (_renderEngine.compare("ogre2") != 0)
+  {
+    igndbg << "Engine '" << _renderEngine
+      << "' doesn't support segmentation cameras" << std::endl;
+    return;
+  }
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "segmentation_camera");
+
+  ignition::sensors::SegmentationCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::SegmentationCameraSensor>(sdfSensor);
+
+  ASSERT_NE(sensor, nullptr);
+  sensor->SetScene(scene);
+
+  EXPECT_EQ(width, (int)sensor->ImageWidth());
+  EXPECT_EQ(height, (int)sensor->ImageHeight());
```

Comment:
```suggestion
  EXPECT_EQ(width, static_cast<int>(sensor->ImageWidth()));
  EXPECT_EQ(height, static_cast<int>(sensor->ImageHeight()));
```
nit: use static_cast instead of c-style cast: https://en.cppreference.com/w/cpp/language/static_cast